smarts/core/utils/logging.py

```diff
@@ -37,7 +37,7 @@ def timeit(name: str, logger):
 def isnotebook():
     try:
         shell = get_ipython().__class__.__name__
-        if shell == "ZMQInteractiveShell":
+        if shell == "ZMQInteractiveShell" or "google.colab" in sys.modules:
```

Comment:
Does is this check for "google.colab" need to be after `ipython`.

Reply:
`"google.colab" in sys.modules` is to detect whether the code is being executed in Colab.